### ISSUE TITLE
improve grouping with self-loops

### ISSUE THREAD

**Issue by adisos** [np-guard/vpc-network-config-analyzer#98]
```suggestion
vsi3a-ky[10.240.30.5],vsi3b-ky[10.240.30.6],vsi3c-ky[10.240.30.4] => vsi3a-ky[10.240.30.5],vsi3b-ky[10.240.30.6],vsi3c-ky[10.240.30.4] : All Connections
vsi3a-ky[10.240.30.5],vsi3b-ky[10.240.30.6],vsi3c-ky[10.240.30.4] => vsi1-ky[10.240.10.4] : All Connections
```
This requires adding a self loop to each VSI, run the grouping algorithm, then remove left-over self loops.
Can be handled in a future PR.

_Originally posted by @zivnevo in https://github.com/np-guard/vpc-network-config-analyzer/pull/92#discussion_r1253989708_

**Comment by ShiriMoran** [comment]
> ```
> This requires adding a self loop to each VSI, run the grouping algorithm, then remove left-over self loops.
> Can be handled in a future PR.
> 
> _Originally posted by @zivnevo in https://github.com/np-guard/vpc-network-config-analyzer/pull/92#discussion_r1253989708_
>             
> ```

@zivnevo adding a self loop before running the algorithm seems wrong.
Consider subnets `a,b,c,d` with the following connections:
```
a => b
a => c
d => c
d => b
```
With the current algorithm, after grouping src we'll get
```
a => b,c
d => b,c
```
and after grouping dst on the result we'll get
`a,d => b,c`

but if we add self loops before running the alg after grouping src we'll get:
```
a => a,b,c
d => b,c,d
```
and grouping dst will group nothing.

I'm working on a different solution.

**Comment by ShiriMoran** [comment]
Actually I'm afraid what is really required is to check all the possible combinations of with and without self loops, which is not reasonable complexity rise. I have another solution in mind, if you wish we can discuss or I can put it here before I write the code.

**Comment by ShiriMoran** [comment]
Proposed algorithm:
* execute the current grouping algorithm for grouping, assume w.l.o.g we grouped destinations 
* Merge lines whose distance is zero until reaching a fix point.

The distance between a line 1 with src `a_1` and dst `b_2`  and line 2 with src `a_2` and distance `b_2` is defined as follows:

**Comment by ShiriMoran** [comment]
solved by #152
